docs/specs/mlnet-database-loader/mlnet-database-loader-specs.md

```diff
@@ -0,0 +1,407 @@
+        - Other relational databases such as MySQL, Oracle, PostgreSQL, etc.
+
+- Scope not supported in this feature:
+    - No-SQL Databases, such as:
+        - Azure Cosmos DB
+        - MongoDB
+        - Any other non-SQL database
+    - Big Data, such as:
+        - Big Data: Spark, Hadoop, etc.
+
+
+## Evidence
+
+Sample feedback and quotes:
+
+- *"[Rachel Poulsen comment](https://www.quora.com/What-are-the-most-valuable-skills-to-learn-for-a-data-scientist-now), Data Scientist at Silicon Valley Data Science: I have been a Data Scientist in Silicon Valley for the last 4 years. There are A LOT of tools out there for data science as I'm sure you are aware. But very importantly, SQL: I'd argue this is 100% necessary for all data scientists, regardless of whether you are using structured or unstructured data. Too many people and companies still use it and are comfortable with it for it to be thrown out with NoSQL techniques. Even companies using NoSQL databases and unstructured data are still using SQL."*
+
+- *"forki commented on May 17, 2018
+I did not say EF ;-) - I heard some of the authors of this package worked on EF before, but please please don't bind it to EF, or sql server or any other specific tech. Or course you want to get data into the learning system without going over to csv. But the data is usually already flattened anyway, so no need for a complex OR mapping tool - especially since the target data structure is Immutable"*
+
+- *Other customers - TBD*
+
+# Who is the Customer
+
+- .NET developers used to work with databases who are getting started with machine learning with ML.NET.
+
+
+# Goals
+
+The business goals are the following, depending on the possible scenarios:
+
+- Ability for developers to load and automatically stream data from relational databases in order to train/evaluate ML.NET models.
+- The code to load from a database should be extremely easy, a single line of code in most cases.
+- Tooling (Model Builder in VS and the CLI) and AutoML API should also support this feature.  
+
+# Solution
+
+The solution is to create an ML.NET database loader classes supporting the above scenarios. 
+
+The main supported features are:
+
+- **Easily load data into IDataView from databases**: The main goal is to load data into IDataView from databases very easily (a single line of code as one of the approaches) and properly (with good performance when streaming data while training).
+
+- **Train ML.NET models with data streaming coming from a database**: As the ultimate goal, the main purpose of loading data into an IDataView is to usually train an ML.NET model with data coming from a database table (or view or SQL sentence).
+
+- **Evaluate ML.NET models with data streaming coming from a database**: As secondary goal is to evaluate the quality/accuracy of an ML.NET model with data coming from a database table (or view or SQL sentence) by comparing it with a set or predictions. This goal should support:
```

Comment:
This includes the feature where I can score data in a DBMS using ML.NET, right?

Reply:
Not initially, as I explained in the previous comment I initially think that should be a different feature in a different specs docs.

I'm guessing you mean [SQL Server CLR integration](https://docs.microsoft.com/en-us/sql/relational-databases/clr-integration/clr-integration-overview?view=sql-server-2017), right?

The implementation of that approach is initially very different and very particular per RDBMS. Even regular Azure SQL Database (single databases and elastic pools) don't support that, only 'SQL Server on-premises' and 'Azure SQL Database managed instance'.

Then, for other RDBMS such a feature is either using a very different implementation or not supported for the CLR.

But, as mentioned, I find it particularly useful for intense operations that need to be 'close' to the data, such as when training a model. :)

Reply:
I updated the specs docs so it includes cases such as [SQL Server CLR integration](https://docs.microsoft.com/en-us/sql/relational-databases/clr-integration/clr-integration-overview?view=sql-server-2017) as part of the design.